### ISSUE TITLE
Add the item to clear the history

### ISSUE THREAD

**Issue by wvxwxvw** [stephanmahieu/formhistorycontrol-2#69]
Don't forget about privacy. I was surprised to find that there was no way to clear history. How so? =)

**Comment by stephanmahieu** [comment]
There are multiple ways to clear the form history stored by this addon. You can either use the [Cleanup](https://stephanmahieu.github.io/fhc-home/Manual/manual/#clean-up) option (automatic and manual) or simply [select and delete](https://stephanmahieu.github.io/fhc-home/Manual/manual/#the-main-dialog) one or multiple (all) items.

Note that for this (or any other plugin) there is no way to clear the history that is maintained by the browser itself. You can however enable the [Override Autocomplete](https://stephanmahieu.github.io/fhc-home/Manual/manual/#miscellaneous) option to prevent the browser from storing history.

**Comment by wvxwxvw** [comment]
There is no way to delete the history in your add-on.

1. In the settings, you can not delete everything, but only everything older than one day.
2. You didn't implement multiple selection in the list, so you won't be able to delete everything.

Clearing items one by one!!! I don't even know why you're offering this to me. I have hundreds of entries accumulated in a few hours per session. What do you suggest I do, every time before closing the browser, click on each item twice?

I'm not asking for advice on how to get out of my skin to delete the history of forms. I ask for an option in the settings, simple and without frills.

Understand that in some places, in some circumstances, the preservation of history can cost freedom. Now compare your advice and the freedom of man! I do not risk, but however I also do not like to leave any history after the session and the lack of complete cleaning makes me delete the extension until better times, perhaps you will understand and change your mind.

**Comment by stephanmahieu** [comment]
This add-on does not support sessions you may have on a page. So cleaning up after a session is not possible. Further more only multiline fields are (by design) associated with a page (url).

If you wish to retain your data only for a short period you can use the cleanup option and set the retention period to 1 day.

**Comment by wvxwxvw** [comment]
I meant browser sessions.

**Comment by stephanmahieu** [comment]
So clear all history when the browser exits?

**Comment by wvxwxvw** [comment]
Yes, when I exit, my history is cleared standard browser tools. I would like that the add-on would have the ability to clean up after a normal exit.

**Comment by wvxwxvw** [comment]
And, in addition, still simplify access to the function of clearing the entire history. Open a dialog, open the context menu, select all, and click delete. This is too long an algorithm for such a simple and obvious function. You have a context menu of the button, which has an excess of extra items, why not add a "full cleaning" item?

**Comment by stephanmahieu** [comment]
Unfortunately there is no shutdown hook available in the WebExtensions API to perform actions just prior to shutdown.

**Comment by wvxwxvw** [comment]
And to delete a folder *^firstPartyDomain=*, through API too it is impossible?

You at least add the usual full cleaning. I temporarily put this folder in the script. But I use it (the script) only before the profile backup. Need a standard full cleaning, on easily accessible button, not somewhere there, for thirty clicks.

**Comment by stephanmahieu** [comment]
No the WebExtension API, by design, does not support any method to interact directly with files or folders on the host OS. The framework/API available to extensions is designed with security, privacy, reliability and stability in mind.

It is possible to create extensions that use native code designed for a specific OS but they fall in a different category and that is not the way I want to go.

**Comment by wvxwxvw** [comment]
Well, let's at least a button and an item in the context menu of the extension. Can be done with confirmation.

**Comment by stephanmahieu** [comment]
Check 👍

**Comment by wvxwxvw** [comment]
I have not found.

**Comment by stephanmahieu** [comment]
Patience....
